code/modules/mob/living/carbon/xenomorph/XenoOverwatch.dm

```diff
@@ -71,10 +71,10 @@
 
 		SEND_SIGNAL(src, COMSIG_XENO_STOP_OVERWATCH, oldXeno)
 		UnregisterSignal(src, COMSIG_MOB_MOVE_OR_LOOK)
-		UnregisterSignal(oldXeno, COMSIG_PARENT_QDELETING)
 
 		if(oldXeno)
 			to_chat(src, SPAN_XENOWARNING("You stop watching [oldXeno]."))
+			UnregisterSignal(oldXeno, COMSIG_PARENT_QDELETING)
```

Comment:
Honestly should just also stick it under the !QDELETED check too since there already is one being performed. QDEL should already unregister it, so there's the possibility of it double unregistering currently.